services/src/main/java/cz/muni/fi/pa165/services/UserServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package cz.muni.fi.pa165.services;
+
+import java.util.List;
+
+import org.springframework.security.crypto.bcrypt.BCrypt;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import cz.muni.fi.pa165.dao.UserDao;
+import cz.muni.fi.pa165.entity.Users;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Inject
+    private UserDao userDao;
+
+
+    @Override
+    public Long create(Users user) {
+        userDao.create(BcryptHashing(user));
+        return user.getId();
+    }
+
+    @Override
+    public Users findById(Long id) {
+        return userDao.findById(id);
+    }
+
+    @Override
+    public List<Users> findAll() {
+        return userDao.findAll();
+    }
+
+    @Override
+    public Users findByName(String name) {
+        return userDao.findByName(name);
+    }
+
+    @Override
+    public void remove(Users user) {
+        userDao.remove(user);
+    }
+
+    @Override
+    public void update(Users user) {
+        // change password
+        if (authenticate(user,findById(user.getId()))){
+                //then hashing password
+            BcryptHashing(user);}
+        userDao.update(user);
+    }
+
+    public boolean  authenticate(Users originalUser,Users  generatedSecuredUser){
+        //boolean matched = BCrypt.checkpw(originalPassword, generatedSecuredPasswordHash);
+        //boolean semaphore = BCrypt.checkpw(oldUser.getPassword(), newUser.getPassword());
+        return  BCrypt.checkpw(originalUser.getPassword(), generatedSecuredUser.getPassword());
+    }
+
+    public Users  BcryptHashing(Users user){
```

Comment:
Osobně bych spíš použil buď PBKDV2 nebo Scrypt, bcrypt má ne úplně hezký vlastnosti, jakmile se vstup dostane nad 70 bytů, to rozhraní není o moc složitější, viz. https://docs.spring.io/spring-security/site/docs/4.2.5.RELEASE/apidocs/org/springframework/security/crypto/scrypt/SCryptPasswordEncoder.html A mimochodem, proč byla použita verze z roku 2011, kde jsi fakt nic lepšího než ten bcrypt použít nemoh - nic lepšího tam totiž není?

Reply:
Tohle jsem nejak neresil. Kazdopadne pokud budeme chtit pouzit PBKDF2, tak ne tu implementaci, co jsem posilal, ale tohle:
https://docs.spring.io/spring-security/site/docs/4.2.8.RELEASE/apidocs/org/springframework/security/crypto/password/Pbkdf2PasswordEncoder.html

Opravdu nechci videt array bytu

Reply:
Jasný. Jelikož u mě u tohohle pr ale už nic nezbylo, nebo jsem si všim jen už nahlášenejch věcí, tak jsem to řešil. A souhlas, to PasswordEncoder interface je mnohem hezčí, než pole bytů, s tím souhlasím.